modules/katamari/src/main/ts/ephox/katamari/api/Singleton.ts

```diff
@@ -1,72 +1,66 @@
 import { Option } from './Option';
 import { Cell } from './Cell';
 
-const revocable = function <T> (doRevoke: (data: T) => void) {
+interface Singleton<T> {
+  clear: () => void;
+  isSet: () => boolean;
+  set: (value: T) => void;
+}
+
+interface Api<T> extends Singleton<T> {
+  run: (fn: (data: T) => void) => void;
+}
+
+export interface Value<T> extends Singleton<T> {
+  on: (fn: (data: T) => void) => void;
+}
+
+const revocable = <T> (doRevoke: (data: T) => void): Singleton<T> => {
   const subject = Cell(Option.none<T>());
 
-  const revoke = function () {
-    subject.get().each(doRevoke);
-  };
+  const revoke = (): void => subject.get().each(doRevoke);
 
-  const clear = function () {
+  const clear = () => {
     revoke();
     subject.set(Option.none());
   };
 
-  const set = function (s: T) {
+  const isSet = () => subject.get().isSome();
+
+  const set = (s: T) => {
     revoke();
     subject.set(Option.some(s));
   };
 
-  const isSet = function () {
-    return subject.get().isSome();
-  };
-
   return {
     clear,
     isSet,
     set
   };
 };
 
-export const destroyable = function <T extends { destroy: () => void }> () {
-  return revocable<T>(function (s) {
-    s.destroy();
-  });
-};
+export const destroyable = <T extends { destroy: () => void }> (): Singleton<T> => revocable<T>((s) => s.destroy());
```

Comment:
This probably needs an exported type, otherwise you can't pass it around (similar to the issue you had with value). So it might be good to do something like this above
```
export interface Revocable<T> extends Singleton<T> { }
```
and then change these to return the `Revocable` type.

Reply:
Any reason I shouldn't just export `Singleton<T>`?

Reply:
It'd be confusing about which type to use.

Reply:
Another reason would be this allows you to add items to just `revocable` without having to do a breaking change (as you'd have to add a new type later).